docs/extensions/guides/ipc.md

```diff
@@ -40,38 +40,46 @@ The general terminology is as follows:
 
 To register either a handler or a listener, you should do something like the following:
 
+File1:
 ```typescript
 import { LensMainExtension, Interface, Types } from "@k8slens/extensions";
 
 export class ExampleExtensionMain extends LensMainExtension {
   onActivate() {
-    this.disposers.push(
-      this.listenIpc({
-        channel: "initialize",
-        listener: this.initializeListener,
-        verifier: this.initializeVerifier,
-      })
-    );
+    this.listenIpc({
+      channel: "initialize",
+      listener: this.initializeListener,
+      verifier: this.initializeVerifier,
+    });
   }
 
-  initializeListener = (event: Types.IpcMainEvent, uid: string) => {
-    console.log(`starting to initialize: ${uid}`);
+  initializeListener = (event: Types.IpcMainEvent, id: string) => {
+    console.log(`starting to initialize: ${id}`);
   };
 
-  initializeVerifier = (args: unknown[]): args is [uid: string] => {
+  initializeVerifier = (args: unknown[]): args is [id: string] => {
     return args.length === 1 && typeof args[0] === "string";
   }
 }
 ```
 
-If you want to register a "handler" you would call `this.handleIpc(...)` instead.
+File2:
+```typescript
+import { LensMainExtension, Interface, Types } from "@k8slens/extensions";
 
-The `LensExtension.prototype.disopsers` is a list of `() => void`'s (or `Utils.Disposer`'s).
-Those functions are sort of like destructors in that they should clean up some state.
-Some functions return them to indicate that it will clean up the state which was just registered.
+export class ExampleExtensionRenderer extends LensRendererExtension {
+  onActivate() {
+    setTimeout(() => this.sendIpc("initialize", "an-id"), 5000);
+  }
+}
+```
+
+As this example shows: the channel names *must* be the same.
+It should also be noted that "listeners" and "handlers" are specific to either `LensRendererExtension` and `LensMainExtension`.
+There is no behind the scenes transfer of these functions.
```

Comment:
So from this, my understanding is that a listener is not a two-way channel or a two-way pipe. It's for one side or the other to declare they're just listening, and you use this when, in this case, as `main`, you just want to be notified of some event, and the `renderer` here notifies without waiting for a response.

And if I wanted the opposite to happen, where `main` notifies `renderer` of something, then in `LensRendererExtension`, I would declare this:

```javascript
this.listenIpc({
  channel: "initialize",
  listener: this.initializeListener,
  verifier: this.initializeVerifier,
});
```

Now would this cause problems if I use the same channel name?

Reply:
Yes that is correct, it is not a two way channel.

If you want `renderer` to wait for a response from `main`. Then on `main` you should use `this.handleIpc()` and on `renderer` you should use `await this.invokeIpc()`. (I think that is in the docs).

I guess I should probably change `this.sendIpc()` to `this.broadcastIpc()` because that is what it is doing. It is sending to all frames and `main`.

So I would recommend using a different name if you want to have `main` be the source.

Will add some items to the docs about this.

Reply:
Sounds good.

Reply:
> I guess I should probably change this.sendIpc() to this.broadcastIpc() because that is what it is doing. It is sending to all frames and main.

SGTM.